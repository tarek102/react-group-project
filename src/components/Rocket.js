/* eslint-disable camelcase, no-unused-vars, react/prop-types, react/jsx-no-undef, no-undef */

import React from 'react';
import { useDispatch } from 'react-redux/es/exports';
import { PropTypes } from 'prop-types';
import { rocketBook, rocketCancel } from '../redux/rockets/rockets';

const Rocket = (props) => {
  const dispatch = useDispatch();

  const {
    id,
    rocket_name,
    description,
    flickr_images,
    reserved,
  } = props;

  return (
    <div id={id} key={id} className="d-flex flex-wrap flex-lg-nowrap m-2 p-2">
      <img src={flickr_images} width="250" height="180" className="image mx-2" alt="rocket-one" />
      <div className="wrapper">
        <h2 className="rocketName">{rocket_name}</h2>
        <div className="description">
          {reserved && <h4>Reserved</h4>}
          {description}
        </div>
        {reserved ? (
          <button
            id={id}
            className="cancelBtn btn btn-outline-danger"
            type="button"
            onClick={() => dispatch(rocketCancel(id))}
          >
            Cancel Reserve
          </button>
        ) : (
          <button
            id={id}
            className="reserveBtn btn btn-outline-info"
            type="button"
            onClick={() => dispatch(rocketBook(id))}
          >
            Reserve Rocket
          </button>
        )}
      </div>
    </div>
  );
};

Rocket.propTypes = {
  id: PropTypes.string.isRequired,
  rocket_name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  flickr_images: PropTypes.string.isRequired,
};
export default Rocket;
