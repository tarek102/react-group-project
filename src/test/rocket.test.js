/* eslint-disable import/no-unresolved */

import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Rocket from '../components/Rocket';

it('missions display correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Rocket />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
