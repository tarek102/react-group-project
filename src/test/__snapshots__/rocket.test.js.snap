// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`missions display correctly 1`] = `
<div
  className="d-flex flex-wrap flex-lg-nowrap my-2 py-2"
>
  <img
    alt="rocket-one"
    className="image mx-2"
    height="220"
    width="290"
  />
  <div
    className="wrapper d-flex flex-column justify-content-start h-5"
  >
    <h2
      className="rocketName"
    />
    <div
      className="description my-3"
    />
    <button
      className="reserveBtn btn btn-outline-info w-25"
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</div>
`;
